pkgs/build-support/fetchurl/default.nix

```diff
@@ -1,4 +1,4 @@
-{ stdenv, curl }: # Note that `curl' may be `null', in case of the native stdenv.
+{ stdenv, curl, ca_cert_file ? null, insecure ? false }: # Note that `curl' may be `null', in case of the native stdenv.
```

Comment:
I see. I suspect the main part of performance loss is due to extra latency of https handshake, but according to docs it seems that would happen *even* with ``--insecure`` with http**s**, so it's more about whether to prefer https URLs over http ones.

I already thought before: it would be nice if fetching was done over plain http/ftp/... by default, but failed-hash ones would be re-checked securely afterward (ideally by PGP, or at least https re-fetch).